src/convert_image/sse2.rs

```diff
@@ -267,6 +267,46 @@ unsafe fn lrgb_to_yuv_4x(
     );
 }
 
+unsafe fn lrgb_to_i420_4x(
+    rgb0: *const u8,
+    rgb1: *const u8,
+    y0: *mut u8,
+    y1: *mut u8,
+    u: *mut u8,
+    v: *mut u8,
+    sampler: Sampler,
+    y_weigths: &[__m128i; 3],
+    uv_weights: &[__m128i; 3],
+) {
+    let (rg0, bg0) = unpack_ui8x3_i16x2_4x(rgb0, sampler);
+    pack_i32_4x(
+        y0,
+        fix_to_i32_4x!(affine_transform(rg0, bg0, y_weigths), FIX16),
+    );
+
+    let (rg1, bg1) = unpack_ui8x3_i16x2_4x(rgb1, sampler);
+    pack_i32_4x(
+        y1,
+        fix_to_i32_4x!(affine_transform(rg1, bg1, y_weigths), FIX16),
+    );
+
+    let srg = sum_i16x2_neighborhood_2x(rg0, rg1);
+    let sbg = sum_i16x2_neighborhood_2x(bg0, bg1);
+
+    let fixed = fix_to_i32_4x!(affine_transform(srg, sbg, uv_weights), FIX18);
+
+    // shuff: ******v1 ******v0 ******u1 ******u0
+    let shuff = _mm_shuffle_epi32(fixed, mm_shuffle(3, 1, 2, 0));
+
+    // uv_res: v1v0u1u0
+    let packed_to_32 = _mm_packs_epi32(shuff, shuff);
```

Comment:
You can create a function to wrap:

```rust
    /// Convert int to char (4-wide)
    ///
    /// x:        ******x3 ******x2 ******x1 ******x0
    /// result:   x3x2x1x0
    unsafe fn u32_to_u8_4x(x: __m128i) -> u32 {
        // Add the three conversion instructions
    }
```

Then have the following:

```rust
let uv = u32_to_u8_4x(u0u1v0v1);
```

Reply:
Since this code is specific to this case i would avoid to create this extra method, since it is only used here